sdk/cosmos/azure-cosmos/src/test/java/com/azure/cosmos/implementation/directconnectivity/ConnectionStateListenerTest.java

```diff
@@ -0,0 +1,96 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.cosmos.implementation.directconnectivity;
+
+import com.azure.cosmos.DirectConnectionConfig;
+import com.azure.cosmos.implementation.Configs;
+import com.azure.cosmos.implementation.ConnectionPolicy;
+import com.azure.cosmos.implementation.Document;
+import com.azure.cosmos.implementation.OperationType;
+import com.azure.cosmos.implementation.ResourceType;
+import com.azure.cosmos.implementation.RxDocumentServiceRequest;
+import com.azure.cosmos.implementation.UserAgentContainer;
+import com.azure.cosmos.implementation.directconnectivity.TcpServerMock.TcpServerFactory;
+import com.azure.cosmos.implementation.directconnectivity.TcpServerMock.TcpServer;
+import com.azure.cosmos.implementation.directconnectivity.TcpServerMock.RequestResponseType;
+import com.azure.cosmos.implementation.directconnectivity.TcpServerMock.SslContextUtils;
+import com.azure.cosmos.implementation.routing.PartitionKeyRangeIdentity;
+import io.netty.handler.ssl.SslContext;
+import org.mockito.Mockito;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+
+import java.util.HashMap;
+import java.util.UUID;
+
+import static com.azure.cosmos.implementation.TestUtils.mockDiagnosticsClientContext;
+
+public class ConnectionStateListenerTest {
+
+    private static int port = 8082;
+    private static String serverUriString = "rntbd://localhost:" + port;
+
+    @DataProvider(name = "connectionStateListenerConfigProvider")
+    public Object[][] connectionStateListenerConfigProvider() {
+        return new Object[][]{
+            // isTcpConnectionEndpointRediscoveryEnabled, serverResponseType, GlobalAddressResolver.updateAddresses() called times
+            {true, RequestResponseType.CHANNEL_FIN, 1},
+            {false, RequestResponseType.CHANNEL_FIN, 0},
+            {true, RequestResponseType.CHANNEL_RST, 0},
+            {false, RequestResponseType.CHANNEL_RST, 0},
+        };
+    }
+
+    @Test(dataProvider = "connectionStateListenerConfigProvider")
```

Comment:
please add `groups =  { "unit" }` similar to other tests.

Reply:
Added.